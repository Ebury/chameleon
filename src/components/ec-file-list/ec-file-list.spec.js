import { mount } from '@vue/test-utils';

import EcFileList from './ec-file-list.vue';

describe('EcFileList', () => {
  function mountFileList(props, mountOpts) {
    return mount(EcFileList, {
      props,
      ...mountOpts,
    });
  }

  const items = [
    { name: 'My invoice.pdf' },
    { name: 'This is a very long file name that is for testing the ellipses.pdf' },
    { name: 'DOC_123123_2423490802348_12312323.pdf' },
    { name: 'number_rates_trades.xscl' },
    { name: 'untitled.pdf' },
  ];

  describe(':props', () => {
    it('should not render any items if nothing is passed', () => {
      const wrapper = mountFileList();
      expect(wrapper.element).toMatchSnapshot();
    });

    it('should render correctly if items are passed', () => {
      const wrapper = mountFileList({ items });
      expect(wrapper.element).toMatchSnapshot();
    });

    it('should render the delete button as disabled if delete disabled is true', () => {
      const wrapper = mountFileList({ items, isDeleteDisabled: true });
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('@events', () => {
    it('should emit deleted item', async () => {
      const wrapper = mountFileList({ items });
      await wrapper.findByDataTest('ec-file-list__item--1').findByDataTest('ec-file-list__delete-btn--1').trigger('click');
      expect(wrapper.emitted('delete').length).toBe(1);
    });

    it('should not emit a deleted item if delete disabled is true', async () => {
      const wrapper = mountFileList({ items, isDeleteDisabled: true });
      await wrapper.findByDataTest('ec-file-list__item--1').findByDataTest('ec-file-list__delete-btn--1').trigger('click');
      expect(wrapper.emitted('delete')).toBeUndefined();
    });
  });
});
