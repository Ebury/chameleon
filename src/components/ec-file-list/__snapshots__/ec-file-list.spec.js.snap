// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EcFileList :props should not render any items if nothing is passed 1`] = `
<div
  class="ec-file-list"
  data-test="ec-file-list"
/>
`;

exports[`EcFileList :props should render correctly if items are passed 1`] = `
<div
  class="ec-file-list"
  data-test="ec-file-list"
>
  <div
    class="ec-file-list__item"
    data-test="ec-file-list__item ec-file-list__item--0"
  >
    <svg
      class="ec-file-list__document-icon ec-icon ec-icon--interactive"
      data-test="ec-file-list__document-icon"
      height="20"
      width="20"
    >
      <use
        xlink:href="#ec-simple-document"
      />
    </svg>
     
    <span
      class="ec-file-list__name"
      data-test="ec-file-list__name ec-file-list__name--0"
      title="My invoice.pdf"
    >
      
      My invoice.pdf
    
    </span>
     
    <button
      class="ec-btn ec-file-list__btn ec-btn--md"
      data-test="ec-file-list__btn--0 ec-btn"
      type="button"
    >
      <svg
        class="ec-btn__icon ec-icon ec-btn__icon--with-spacing"
        data-test="ec-btn__icon"
        height="22"
        width="22"
      >
        <use
          xlink:href="#ec-simple-outline-delete"
        />
      </svg>
       
      <span
        class="ec-btn__text"
        data-test="ec-btn__text"
      >
        <span
          class="tw-sr-only"
        >
          Delete document
        </span>
      </span>
       
      <!---->
    </button>
  </div>
  <div
    class="ec-file-list__item"
    data-test="ec-file-list__item ec-file-list__item--1"
  >
    <svg
      class="ec-file-list__document-icon ec-icon ec-icon--interactive"
      data-test="ec-file-list__document-icon"
      height="20"
      width="20"
    >
      <use
        xlink:href="#ec-simple-document"
      />
    </svg>
     
    <span
      class="ec-file-list__name"
      data-test="ec-file-list__name ec-file-list__name--1"
      title="This is a very long file name that is for testing the ellipses.pdf"
    >
      
      This is a very long file name that is for testing the ellipses.pdf
    
    </span>
     
    <button
      class="ec-btn ec-file-list__btn ec-btn--md"
      data-test="ec-file-list__btn--1 ec-btn"
      type="button"
    >
      <svg
        class="ec-btn__icon ec-icon ec-btn__icon--with-spacing"
        data-test="ec-btn__icon"
        height="22"
        width="22"
      >
        <use
          xlink:href="#ec-simple-outline-delete"
        />
      </svg>
       
      <span
        class="ec-btn__text"
        data-test="ec-btn__text"
      >
        <span
          class="tw-sr-only"
        >
          Delete document
        </span>
      </span>
       
      <!---->
    </button>
  </div>
  <div
    class="ec-file-list__item"
    data-test="ec-file-list__item ec-file-list__item--2"
  >
    <svg
      class="ec-file-list__document-icon ec-icon ec-icon--interactive"
      data-test="ec-file-list__document-icon"
      height="20"
      width="20"
    >
      <use
        xlink:href="#ec-simple-document"
      />
    </svg>
     
    <span
      class="ec-file-list__name"
      data-test="ec-file-list__name ec-file-list__name--2"
      title="DOC_123123_2423490802348_12312323.pdf"
    >
      
      DOC_123123_2423490802348_12312323.pdf
    
    </span>
     
    <button
      class="ec-btn ec-file-list__btn ec-btn--md"
      data-test="ec-file-list__btn--2 ec-btn"
      type="button"
    >
      <svg
        class="ec-btn__icon ec-icon ec-btn__icon--with-spacing"
        data-test="ec-btn__icon"
        height="22"
        width="22"
      >
        <use
          xlink:href="#ec-simple-outline-delete"
        />
      </svg>
       
      <span
        class="ec-btn__text"
        data-test="ec-btn__text"
      >
        <span
          class="tw-sr-only"
        >
          Delete document
        </span>
      </span>
       
      <!---->
    </button>
  </div>
  <div
    class="ec-file-list__item"
    data-test="ec-file-list__item ec-file-list__item--3"
  >
    <svg
      class="ec-file-list__document-icon ec-icon ec-icon--interactive"
      data-test="ec-file-list__document-icon"
      height="20"
      width="20"
    >
      <use
        xlink:href="#ec-simple-document"
      />
    </svg>
     
    <span
      class="ec-file-list__name"
      data-test="ec-file-list__name ec-file-list__name--3"
      title="number_rates_trades.xscl"
    >
      
      number_rates_trades.xscl
    
    </span>
     
    <button
      class="ec-btn ec-file-list__btn ec-btn--md"
      data-test="ec-file-list__btn--3 ec-btn"
      type="button"
    >
      <svg
        class="ec-btn__icon ec-icon ec-btn__icon--with-spacing"
        data-test="ec-btn__icon"
        height="22"
        width="22"
      >
        <use
          xlink:href="#ec-simple-outline-delete"
        />
      </svg>
       
      <span
        class="ec-btn__text"
        data-test="ec-btn__text"
      >
        <span
          class="tw-sr-only"
        >
          Delete document
        </span>
      </span>
       
      <!---->
    </button>
  </div>
  <div
    class="ec-file-list__item"
    data-test="ec-file-list__item ec-file-list__item--4"
  >
    <svg
      class="ec-file-list__document-icon ec-icon ec-icon--interactive"
      data-test="ec-file-list__document-icon"
      height="20"
      width="20"
    >
      <use
        xlink:href="#ec-simple-document"
      />
    </svg>
     
    <span
      class="ec-file-list__name"
      data-test="ec-file-list__name ec-file-list__name--4"
      title="untitled.pdf"
    >
      
      untitled.pdf
    
    </span>
     
    <button
      class="ec-btn ec-file-list__btn ec-btn--md"
      data-test="ec-file-list__btn--4 ec-btn"
      type="button"
    >
      <svg
        class="ec-btn__icon ec-icon ec-btn__icon--with-spacing"
        data-test="ec-btn__icon"
        height="22"
        width="22"
      >
        <use
          xlink:href="#ec-simple-outline-delete"
        />
      </svg>
       
      <span
        class="ec-btn__text"
        data-test="ec-btn__text"
      >
        <span
          class="tw-sr-only"
        >
          Delete document
        </span>
      </span>
       
      <!---->
    </button>
  </div>
</div>
`;

exports[`EcFileList :props should render the delete button as disabled if delete disabled is true 1`] = `
<div
  class="ec-file-list"
  data-test="ec-file-list"
>
  <div
    class="ec-file-list__item"
    data-test="ec-file-list__item ec-file-list__item--0"
  >
    <svg
      class="ec-file-list__document-icon ec-icon ec-icon--interactive"
      data-test="ec-file-list__document-icon"
      height="20"
      width="20"
    >
      <use
        xlink:href="#ec-simple-document"
      />
    </svg>
     
    <span
      class="ec-file-list__name"
      data-test="ec-file-list__name ec-file-list__name--0"
      title="My invoice.pdf"
    >
      
      My invoice.pdf
    
    </span>
     
    <button
      class="ec-btn ec-file-list__btn ec-btn--md"
      data-test="ec-file-list__btn--0 ec-btn"
      disabled="disabled"
      type="button"
    >
      <svg
        class="ec-btn__icon ec-icon ec-btn__icon--with-spacing"
        data-test="ec-btn__icon"
        height="22"
        width="22"
      >
        <use
          xlink:href="#ec-simple-outline-delete"
        />
      </svg>
       
      <span
        class="ec-btn__text"
        data-test="ec-btn__text"
      >
        <span
          class="tw-sr-only"
        >
          Delete document
        </span>
      </span>
       
      <!---->
    </button>
  </div>
  <div
    class="ec-file-list__item"
    data-test="ec-file-list__item ec-file-list__item--1"
  >
    <svg
      class="ec-file-list__document-icon ec-icon ec-icon--interactive"
      data-test="ec-file-list__document-icon"
      height="20"
      width="20"
    >
      <use
        xlink:href="#ec-simple-document"
      />
    </svg>
     
    <span
      class="ec-file-list__name"
      data-test="ec-file-list__name ec-file-list__name--1"
      title="This is a very long file name that is for testing the ellipses.pdf"
    >
      
      This is a very long file name that is for testing the ellipses.pdf
    
    </span>
     
    <button
      class="ec-btn ec-file-list__btn ec-btn--md"
      data-test="ec-file-list__btn--1 ec-btn"
      disabled="disabled"
      type="button"
    >
      <svg
        class="ec-btn__icon ec-icon ec-btn__icon--with-spacing"
        data-test="ec-btn__icon"
        height="22"
        width="22"
      >
        <use
          xlink:href="#ec-simple-outline-delete"
        />
      </svg>
       
      <span
        class="ec-btn__text"
        data-test="ec-btn__text"
      >
        <span
          class="tw-sr-only"
        >
          Delete document
        </span>
      </span>
       
      <!---->
    </button>
  </div>
  <div
    class="ec-file-list__item"
    data-test="ec-file-list__item ec-file-list__item--2"
  >
    <svg
      class="ec-file-list__document-icon ec-icon ec-icon--interactive"
      data-test="ec-file-list__document-icon"
      height="20"
      width="20"
    >
      <use
        xlink:href="#ec-simple-document"
      />
    </svg>
     
    <span
      class="ec-file-list__name"
      data-test="ec-file-list__name ec-file-list__name--2"
      title="DOC_123123_2423490802348_12312323.pdf"
    >
      
      DOC_123123_2423490802348_12312323.pdf
    
    </span>
     
    <button
      class="ec-btn ec-file-list__btn ec-btn--md"
      data-test="ec-file-list__btn--2 ec-btn"
      disabled="disabled"
      type="button"
    >
      <svg
        class="ec-btn__icon ec-icon ec-btn__icon--with-spacing"
        data-test="ec-btn__icon"
        height="22"
        width="22"
      >
        <use
          xlink:href="#ec-simple-outline-delete"
        />
      </svg>
       
      <span
        class="ec-btn__text"
        data-test="ec-btn__text"
      >
        <span
          class="tw-sr-only"
        >
          Delete document
        </span>
      </span>
       
      <!---->
    </button>
  </div>
  <div
    class="ec-file-list__item"
    data-test="ec-file-list__item ec-file-list__item--3"
  >
    <svg
      class="ec-file-list__document-icon ec-icon ec-icon--interactive"
      data-test="ec-file-list__document-icon"
      height="20"
      width="20"
    >
      <use
        xlink:href="#ec-simple-document"
      />
    </svg>
     
    <span
      class="ec-file-list__name"
      data-test="ec-file-list__name ec-file-list__name--3"
      title="number_rates_trades.xscl"
    >
      
      number_rates_trades.xscl
    
    </span>
     
    <button
      class="ec-btn ec-file-list__btn ec-btn--md"
      data-test="ec-file-list__btn--3 ec-btn"
      disabled="disabled"
      type="button"
    >
      <svg
        class="ec-btn__icon ec-icon ec-btn__icon--with-spacing"
        data-test="ec-btn__icon"
        height="22"
        width="22"
      >
        <use
          xlink:href="#ec-simple-outline-delete"
        />
      </svg>
       
      <span
        class="ec-btn__text"
        data-test="ec-btn__text"
      >
        <span
          class="tw-sr-only"
        >
          Delete document
        </span>
      </span>
       
      <!---->
    </button>
  </div>
  <div
    class="ec-file-list__item"
    data-test="ec-file-list__item ec-file-list__item--4"
  >
    <svg
      class="ec-file-list__document-icon ec-icon ec-icon--interactive"
      data-test="ec-file-list__document-icon"
      height="20"
      width="20"
    >
      <use
        xlink:href="#ec-simple-document"
      />
    </svg>
     
    <span
      class="ec-file-list__name"
      data-test="ec-file-list__name ec-file-list__name--4"
      title="untitled.pdf"
    >
      
      untitled.pdf
    
    </span>
     
    <button
      class="ec-btn ec-file-list__btn ec-btn--md"
      data-test="ec-file-list__btn--4 ec-btn"
      disabled="disabled"
      type="button"
    >
      <svg
        class="ec-btn__icon ec-icon ec-btn__icon--with-spacing"
        data-test="ec-btn__icon"
        height="22"
        width="22"
      >
        <use
          xlink:href="#ec-simple-outline-delete"
        />
      </svg>
       
      <span
        class="ec-btn__text"
        data-test="ec-btn__text"
      >
        <span
          class="tw-sr-only"
        >
          Delete document
        </span>
      </span>
       
      <!---->
    </button>
  </div>
</div>
`;
